applications/nrf5340_audio/src/bluetooth/le_audio_bis_gateway.c

```diff
@@ -391,13 +391,27 @@ int le_audio_play_pause(void)
 	return ret;
 }
 
-int le_audio_send(uint8_t const *const data, size_t size)
+int le_audio_send(struct encoded_audio enc_audio)
 {
 	int ret;
 	static bool wrn_printed[CONFIG_BT_AUDIO_BROADCAST_SRC_STREAM_COUNT];
 	struct net_buf *buf;
 	size_t num_streams = ARRAY_SIZE(audio_streams);
-	size_t data_size = size / num_streams;
+	size_t data_size_pr_stream;
+
+	if (enc_audio.num_ch == 1) {
+		data_size_pr_stream = enc_audio.size;
+	} else if (enc_audio.num_ch == num_streams) {
+		data_size_pr_stream = enc_audio.size / num_streams;
+	} else {
+		LOG_ERR("Number of encoded channels must be 1 or equal to the number of streams");
+		return -EINVAL;
+	}
+
+	if (data_size_pr_stream != LE_AUDIO_SDU_SIZE_OCTETS(CONFIG_LC3_BITRATE)) {
```

Comment:
Note that this is assuming that _frames_per_sdu is 1. It is currently hard coded in our presets, but this could and should change to a config